spotify_tensorflow/dataset.py

```diff
@@ -363,10 +363,13 @@ def batch_iterator(cls,
 
         @staticmethod
         def __format_df(batch, multispec_feature_groups):
-            # type: (Dict[str, np.ndarray], Optional[List[List[str]]]) -> Union[pd.DataFrame, List[pd.DataFrame]]  # noqa: E501
+            # type: (Optional[Dict[str, np.ndarray]], Optional[List[List[str]]]) -> Union[pd.DataFrame, List[pd.DataFrame]]  # noqa: E501
             df = pd.DataFrame(batch)
             if not multispec_feature_groups:
-                return df[list(batch.keys())]
+                if batch:
+                    return df[list(batch.keys())]
+                else:
+                    return df[list({})]
```

Comment:
TBH I'm not sure about this, the whole `Optional` thing is to get around an issue with mypy and using `None` to initialize variables...wondering if it's better to just do away with the typing for this case. @ravwojdyla what do you think?